app/src/org/commcare/logic/AndroidFormController.java

```diff
@@ -88,4 +86,14 @@ public FormAnalyticsHelper getFormAnalyticsHelper() {
     public FormDef getFormDef() {
         return mFormEntryController.getModel().getForm();
     }
+
+    // TODO: we should cache this
+    public String getSerializedFormIndex() {
+        try{
+            byte[] serializedFormIndex = SerializationUtil.serialize(getFormIndex());
+            return Base64.encodeToString(serializedFormIndex, Base64.DEFAULT);
+        } catch (Exception e){
+            return null;
```

Comment:
Think it's better to fail fast here instead of swallowing the errors here unless there is a strong reason to do otherwise. At the very least we should be logging the exceptions encountered here.

Reply:
Actually think again it makes sense to catch any errors here as we would not want to loose on saving form because of an error here.